# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../liboil/vfp.patch.arm
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- liboil-0.3.15/liboil/arm/math_vfp.c.old	2008-03-13 20:17:59.000000000 +0000
+++ liboil-0.3.15/liboil/arm/math_vfp.c	2009-03-26 13:53:45.000000000 +0000
@@ -30,7 +30,7 @@
 #include <liboil/liboilclasses.h>
 #include <liboil/liboilfunction.h>
 
-#if __VFP_FP__
+#if defined(__VFP_FP__) && !defined(__SOFTFP__)
 
 extern void vfp_add_f32 (float *d, const float *s1, const float *s2, int n);
 extern void vfp_add_f64 (double *d, const double *s1, const double *s2, int n);
--- liboil-0.3.15/liboil/arm/math_vfp_asm.S.old	2007-11-16 02:53:47.000000000 +0000
+++ liboil-0.3.15/liboil/arm/math_vfp_asm.S	2009-03-26 13:49:33.000000000 +0000
@@ -24,7 +24,7 @@
  * SUCH DAMAGE.
  */
 
-#if __VFP_FP__
+#if defined(__VFP_FP__) && !defined(__SOFTFP__)
 /* 
 ** compile with -mcpu=arm1136j-s -mfpu=vfp -mfloat-abi=softfp
 **

